### PR TITLE
Feature/add to cart

### LINKED ISSUE #14: Create path for Cart.js Component
- [x] Create GET and POST instances for the Cart page in the API

### PR DESCRIPTION
**Pull request by delilahrois**
### What'd ya do to the code?
Added cart functionality. Can select 'Add to Cart' on ImagePage and product will post to the API /cart endpoint. Added conditional logic to check id of product against what is already in the API cart, and update quantity.
Added price and quantity functionality to cart page. UI is updating based on .quantity and .price properties on objects in the API cart.
Subtotal function is working properly and doing math based on the cart in API.

### Select One Thing
- [x] This broke nothing
- [ ] This broke something
- [ ] This broke everything

### Type of change
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor(DRY-ing up code, removing comments etc.)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:
- [ ] If this code needs to be tested, all tests are passing
- [x] The code runs locally
- [ ] There are comments where clarification/ organization is needed
- [x] The code is DRY. If it's not, I tried my best
- [x] I reviewed my code before pushing

### What's next?
Only issue is the bug with each action being 1 behind on the UI-- working on it

### CODE REVIEW

File: src/Components/Cart/Cart.js [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
+      const responseJson = await response.json()
+      setData(responseJson)
+    }
+    fetchData('http://localhost:3001/api/v1/cart')
+  }, [])
+
+
+  const [cartTotal, setCartTotal] = useState('');
+
+  useEffect(() => {
```
Comment (reganlosey) [inline src/Components/Cart/Cart.js:20]: So proud of this function

**Review by reganlosey — APPROVED** [review]
Looks great, we're hot on the tail of this MVP

**Review by edwardkrupicka — APPROVED** [review]
Looks good, great work

File: src/Components/Cart/Cart.js [hunk 2]
```diff
@@ -1,15 +1,41 @@
...
+    const calculateSubTotal = () => {
+      const subTotal = data.reduce((sum, item) => {
+        sum += (item.quantity * parseInt(item.price));
+        return sum;
+      }, 0)
+      setCartTotal(subTotal)
+    }
+    calculateSubTotal()
+  }, [data])
+
```
Comment (tanya2108) [inline src/Components/Cart/Cart.js:30]: Great, a function for subtotal. How are we going to calculate the tax?

**Review by tanya2108 — APPROVED** [review]
Amazing with add to cart functionality and calculating subtotals!

File: src/Components/Cart/Cart.js (hunk 2 above)
Comment (reganlosey) [reply src/Components/Cart/Cart.js:30]: I pulled down and I'm working on the tax and total as we speak!

**Pull request merged**